gradle-palantir-java-format/src/main/java/com/palantir/javaformat/gradle/spotless/NativePalantirJavaFormatStep.java

```diff
@@ -25,41 +25,37 @@
 import java.io.Serializable;
 import java.nio.charset.StandardCharsets;
 import java.util.List;
+import java.util.function.Supplier;
 import org.gradle.api.artifacts.Configuration;
 import org.gradle.api.logging.Logger;
 import org.gradle.api.logging.Logging;
 
 public final class NativePalantirJavaFormatStep {
-    private static Logger logger = Logging.getLogger(NativePalantirJavaFormatStep.class);
+    private static final Logger logger = Logging.getLogger(NativePalantirJavaFormatStep.class);
 
     private NativePalantirJavaFormatStep() {}
 
     private static final String NAME = "palantir-java-format";
 
     /** Creates a step which formats everything - code, import order, and unused imports. */
     public static FormatterStep create(Configuration configuration) {
-        return FormatterStep.createLazy(
-                NAME,
-                () -> {
-                    File execFile = configuration.getSingleFile();
-                    logger.info("Using native-image at {}", configuration.getSingleFile());
-                    return new State(FileSignature.signAsSet(execFile));
-                },
-                State::createFormat);
+        return FormatterStep.createLazy(NAME, () -> new State(configuration::getSingleFile), State::createFormat);
     }
 
     static class State implements Serializable {
         private static final long serialVersionUID = 1L;
 
-        final FileSignature pathToExe;
+        private final transient Supplier<File> execSupplier;
 
-        State(FileSignature pathToExe) {
-            this.pathToExe = pathToExe;
+        State(Supplier<File> supplier) {
+            this.execSupplier = supplier;
         }
 
         String format(ProcessRunner runner, String input) throws IOException, InterruptedException {
-            List<String> argumentsWithPathToExe =
-                    List.of(pathToExe.getOnlyFile().getAbsolutePath(), "--palantir", "-");
+            File execFile = execSupplier.get();
+            logger.info("Using native-image at {}", execFile);
+            File signedFile = FileSignature.signAsSet(execFile).getOnlyFile();
```

Comment:
We still need to have this signature as part of the serialised state otherwise the task will be incorrectly up-to-date. Which means it needs to be a field still that's set at the "correct" time. I think we could make our own `writeObject`  method here to do this at the correct time.

Reply:
Yep, good catch! Let's keep the signature as a field in NativePalantirJavaFormatStep, so up to date checking can occur. 

Let's add a comment to this class in https://github.com/palantir/palantir-java-format/pull/1327/files#r2163566713